db/migrate/20151231125222_extract_access_limiting.rb

```diff
@@ -0,0 +1,21 @@
+class ExtractAccessLimiting < ActiveRecord::Migration
+  def up
+    create_table :access_limits do |t|
+      t.integer :target_id, null: false
+      t.string :target_type, null: false
+      t.json :users, null: false, default: []
+      t.timestamps null: false
+    end
```

Comment:
We could probably do with a unique index across a composite of the target columns.